playbooks/common/openshift-cluster/upgrades/pre/verify_cluster.yml

```diff
@@ -41,6 +41,13 @@
     - openshift_release is defined
     - not (openshift_release is version_compare(openshift_upgrade_target ,'='))
 
+  - fail:
+      msg: >
+        No schedulable masters found, please make sure at least one master doesn't have 'openshift_schedulable=false'
+    when: True not in l_master_schedulable_setting
+    vars:
+      l_master_schedulable_setting: "{{ groups['oo_masters_to_config'] | map('extract', hostvars, 'openshift_schedulable') | map('bool') | list | default([]) }}"
```

Comment:
It's currently not mandatory to set openshift_schedulable and we don't set_fact it at any point.  This variable will be undefined for most hosts, probably.  That looks like the when condition would fail.

Reply:
It won't fail - thanks to `default([])`. Its not explicit though, but ansible filters are not flexible enough to set a default value in `extract`

Reply:
Gah, turns out it worked locally because I had `openshift_schedulable` applied to all hosts in group_vars.

Fixed now - first we get `openshift_schedulable` from hostvars, select all defined vars (this drops all hosts without this var), convert it to bool - and then fail if either a list is empty (no hosts have `openshift_schedulable` set == all schedulable) or all are explicitly marked as unschedulable (no 'true' value in the filtered list)